src/main/java/net/openhft/chronicle/bytes/internal/BytesInternal.java

```diff
@@ -3404,7 +3404,8 @@ static class DateCache {
     }
 
     public static <B extends BytesStore<B, U>, U> BytesStore<B, U> failIfBytesOnBytes(BytesStore<B, U> bytesStore) {
-        if (bytesStore instanceof Bytes) {
+        // MappedBytes don't have a backing BytesStore so we have to allow them to be used this way
```

Comment:
Can you please create a separate issue to remove this check and introduce MappedBytesStore somewhere in the future?